hibike/hid.py

```diff
@@ -0,0 +1,102 @@
+import os
+import sys
+
+import usb.core
+import time
+
+
+class HIDWrapper(object):
+    """Handles low level communication over the USB protocol"""
+    VENDOR_ID = 0x2341
+    PRODUCT_ID = 0x8037
+
+    def __init__(self, device, interface_num, endpoint_address):
+        self.device = device
+        self.interface_num = interface_num
+        self.endpoint_address = endpoint_address
+
+        self.device.set_configuration()
+        self.device.reset()
+
+    @classmethod
+    def open_first(cls, vendor_id=VENDOR_ID, product_id=PRODUCT_ID):
+        for args in cls.find_hid_endpoints(vendor_id, product_id):
+            return HIDWrapper(*args)
+
+        raise usb.USBError("Could not find any device (vendor_id={}, product_id={})".format(vendor_id, product_id))
+
+    @classmethod
+    def open_all(cls, vendor_id=VENDOR_ID, product_id=PRODUCT_ID):
+        res = []
+        for args in cls.find_hid_endpoints(vendor_id, product_id):
+            try:
+                res.append(HIDWrapper(*args))
+            except usb.USBError as e:
+                print("WARNING: Skipping device due to error:", e)
```

Comment:
I'm not sure you'll be able to extract any useful information. At this point all we have are the low level pyusb objects. What information do those contain?